### DISCUSSION TITLE
PollingInterval during sleep?

### DISCUSSION THREAD

**Discussion by junkyardhokie** [philosowaffle/peloton-to-garmin#150] [Q&A]
Does the PollingInterval take into account current system time? I'm wondering how the countdown works when Docker is running on a laptop that is closed (i.e., sleeping). Does the countdown timer only increment while the laptop is active? Or does it update based on current time and compare it start time?

**Comment by philosowaffle** [comment]
The polling interval does not look at system time. It basically starts a timer and runs the sync job again when the timer reaches a specified length of time. The timer will only be active/incrementing so long as the process is running.

I'm not sure how aggressive a laptop sleep cycle is in terms of suspending active processes when it sleeps. So long as the docker container (and therefore the p2g process) is still running, the timer will countdown.  But if the container is truly suspended, then the countdown time will be suspended as well.

**Reply by junkyardhokie** [reply]
I'm pretty confident most of the processes are suspended when the laptop (Macbook) sleeps. 

Would be a good feature to add a wall clock based timer vs a process runtime timer.

**Reply by Vicore1** [reply]
Based on what I found [here](https://apple.stackexchange.com/questions/278185/do-terminal-processes-stop-if-mac-sleeps) sleeping will stop basically everything until it wakes. I imagine that it freezes the the timer.  If it ran off a system timer it could be combine with the [scheduler ](https://support.apple.com/guide/mac-help/schedule-mac-desktop-computer-turn-mchlp2266/mac) to auto-wake to allow for Peloton-to garmin to run.  Honestly, that does seem like a lot of effort given you can just set up the mac to not sleep at all or just sleep the screen from the energy saver settings.

**Reply by junkyardhokie** [reply]
It's actually a pretty simple change. Keep track of start time, periodically check system time, and update the elapsed time. Nothing more to it.
